modules/line_descriptor/src/binary_descriptor.cpp

```diff
@@ -2467,6 +2467,11 @@ int BinaryDescriptor::EDLineDetector::EDline( cv::Mat &image, LineChains &lines
           offsetInLineArray = pLineSID[numOfLines];         // line was not accepted, the offset is set back
         }
       }
+      // Avoid array out of range
+      if(numOfLines >= lines.sId.size()) {
+        lines.sId.emplace_back(offsetInLineArray);
```

Comment:
please use `.push_back` as 3.4 branch doesn't have C++11 requirement (should support C++98)

Reply:
OK, done.